clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -256,27 +256,39 @@ public String toString() {
 
     public static class FutureCompletionHandler implements RequestCompletionHandler {
 
+        private long responseCompletionTimeMs;
         private final CompletableFuture<ClientResponse> future;
 
         FutureCompletionHandler() {
-            this.future = new CompletableFuture<>();
+            future = new CompletableFuture<>();
         }
 
-        public void onFailure(final RuntimeException e) {
-            future.completeExceptionally(e);
+        public void onFailure(final long currentTimeMs, final RuntimeException e) {
```

Comment:
I could not find unit tests to verify the setting of the correct completion time in `onFailure()` and `onComplete()`.

Reply:
You are right, I should add one there.